### PR TITLE
Attach kprobes using kprobe_events ABI

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
This PR changes how the kprobes for network tracer are attached, to use debugfs.
Possible after this change in ebpf-manager: https://github.com/DataDog/ebpf-manager/pull/63

### Motivation
We can now get telemetry about kprobe hits and misses.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by brycekahle** [comment]
Is this something we might want to control via config? We don't currently report this telemetry on customer machines, so it would be better to use the perf ABI in that case.

File: pkg/ebpf/config.go [hunk 1]
```diff
@@ -77,14 +80,15 @@ func NewConfig() *Config {
...
+		EnableRuntimeCompiler:            cfg.GetBool(key(spNS, "enable_runtime_compiler")),
+		RuntimeCompilerOutputDir:         cfg.GetString(key(spNS, "runtime_compiler_output_dir")),
+		EnableKernelHeaderDownload:       cfg.GetBool(key(spNS, "enable_kernel_header_download")),
+		KernelHeadersDirs:                cfg.GetStringSlice(key(spNS, "kernel_header_dirs")),
+		KernelHeadersDownloadDir:         cfg.GetString(key(spNS, "kernel_header_download_dir")),
+		AptConfigDir:                     cfg.GetString(key(spNS, "apt_config_dir")),
+		YumReposDir:                      cfg.GetString(key(spNS, "yum_repos_dir")),
+		ZypperReposDir:                   cfg.GetString(key(spNS, "zypper_repos_dir")),
+		AllowPrecompiledFallback:         cfg.GetBool(key(spNS, "allow_precompiled_fallback")),
+		AttachKprobesWithKprobeEventsABI: cfg.GetBool(key(spNS, "attach_kprobes_with_kprobe_events_abi")),
```
Comment (brycekahle) [inline pkg/ebpf/config.go:92]: You need to define `key(spNS, "attach_kprobes_with_kprobe_events_abi")` in `pkg/config/system_probe.py`

**Review by brycekahle — CHANGES_REQUESTED** [review]

**Review by hmahmood — APPROVED** [review]

File: pkg/network/tracer/tracer_test.go [hunk 2]
```diff
@@ -1846,6 +1847,27 @@ func TestRuntimeCompilerEnvironmentVar(t *testing.T) {
 	assert.NotEqual(t, enabled, cfg.AllowPrecompiledFallback)
 }
 
+func TestKprobeAttachWithKprobeEvents(t *testing.T) {
```
Comment (brycekahle) [inline pkg/network/tracer/tracer_test.go:1850]: I think this needs to be in `tracer_linux_test.go`

**Review by brycekahle — CHANGES_REQUESTED** [review]

**Review by brycekahle — APPROVED** [review]
This looks good. We should explore adding a manager-level option as a default, to simplify things.

**Review by brycekahle — APPROVED** [review]

**Comment by usamasaqib** [comment]
> This looks good. We should explore adding a manager-level option as a default, to simplify things.

The manager would default to using the perf ABI incase nothing is specified.

**Review by guyarb — CHANGES_REQUESTED** [review]

**Review by guyarb — APPROVED** [review]
LGTM

**Review by hmahmood — APPROVED** [review]

**Pull request merged**